application/gqlgen/post.go

```diff
@@ -589,3 +589,76 @@ func (r *mutationResolver) RemovePotentialProvider(ctx context.Context, postID,
 
 	return &post, nil
 }
+
+func (r *mutationResolver) MarkRequestAsDelivered(ctx context.Context, postID string) (*models.Post, error) {
+	cUser := models.CurrentUser(ctx)
+
+	var post models.Post
+	if err := post.FindByUUID(postID); err != nil {
+		return nil, domain.ReportError(ctx, err, "MarkRequestAsDelivered.FindPost")
+	}
+
+	extras := map[string]interface{}{
+		"user": cUser.UUID,
+		"post": post.UUID,
+	}
+
+	if cUser.AdminRole != models.UserAdminRoleSuperAdmin && cUser.ID != post.ProviderID.Int {
+		return nil, domain.ReportError(ctx, errors.New("not allowed to deliver request"),
+			"MarkRequestAsDelivered.Unauthorized", extras)
+	}
+
+	if post.Status != models.PostStatusAccepted && post.Status != models.PostStatusReceived {
```

Comment:
Shouldn't the post state machine check these? If not, perhaps it should. Seems like model validation rules leaking out. (Same on the other mutation.)

Reply:
:+1: